website/content/docs/secrets/kubernetes.mdx

```diff
@@ -0,0 +1,339 @@
+    --header "Authorization: Bearer eyJHbGci0iJSUzI1Ni..."
+{
+  "kind": "PodList",
+  "apiVersion": "v1",
+  "metadata": {
+    "resourceVersion": "1624"
+  },
+  "items": []
+}
+```
+
+When the lease expires, we can verify that the token has been revoked.
+```shell-session
+$ curl -sk $(kubectl config view -o 'jsonpath={.clusters[0].cluster.server}')/api/v1/namespaces/test/pods \
+    --header "Authorization: Bearer eyJHbGci0iJSUzI1Ni..."
+{
+  "kind": "Status",
+  "apiVersion": "v1",
+  "metadata": {},
+  "status": "Failure",
+  "message": "Unauthorized",
+  "reason": "Unauthorized",
+  "code": 401
+}
+```
+
+## TTL
+
+Kubernetes service account tokens have a time-to-live (TTL). When a token expires it is
+automatically revoked.
+
+You can set a default (`token_defaul_ttl`) and a maximum TTL (`token_max_ttl`) when
+creating or tuning the Vault role.
+
+```shell-session
+$ vault write kubernetes/roles/my-role \
+    allowed_kubernetes_namespaces="*" \
+    service_account_name="new-service-account-with-generated-token" \
+    token_default_ttl="10m" \
+    token_max_ttl="2h"
+```
+
+You can also set a TTL (`ttl`) when generate the token from the credentials endpoint.
+The TTL of the token will be given the default if not specified (and cannot exceed the
+maximum TTL of the role, if present).
+
+```shell-session
+$ vault write kubernetes/creds/my-role \
+    kubernetes_namespace=test \
+    ttl=20m
+
+Key                        Value
+–--                        -----
+lease_id                   kubernetes/creds/my-role/31d771a6-...
+lease_duration             20m0s
+lease_renwable             false
+service_account_name       default
+service_account_namespace  test
+service_account_token      eyJHbGci0iJSUzI1NiIsImtpZCI6ImlrUEE...
+```
+
+
+You can verify the token's TTL by decoding the JWT token and extracting the `exp` field.
+
+```shell-session
+$ echo 'eyJhbGc...' | cut -d'.' -f2 | base64 -d  | jq -r '.exp|todateiso8601'
+2022-05-20T10:18:05Z
+```
+
+## Automatically Managing Roles and Service Accounts
+
+When configuring the Vault role, we can pass in parameters to specify that we want to
+automatically generate the Kubernetes service account and role binding,
+and optionally generate the Kubernetes role itself.
+
+If we want to configure the Vault role to use a pre-existing Kubernetes role, but generate
+the service account and role binding automatically, we can set the `kubernetes_role_name`
+parameter.
+
+```shell-session
+$ vault write kubernetes/roles/auto-managed-sa-role \
+    allowed_kubernetes_namespaces="test" \
+    kubernetes_role_name="test-role-list-pods"
+```
+
+~> **Note**: Vault's service account will also need access to the resources it is granting
+access to. This can be done for the examples above with `kubectl -n test create rolebinding --role test-role-list-pods --serviceaccount=vault:vault vault-test-role-abilities`.
```

Comment:
Oh this is a nice call out, I hadn't read this when I left the comment above about escalating privileges.

Reply:
I'll see if I can clarify it and link to the information about escalation of privileges.